### PR TITLE
serialization troubleshooting in console

### PR DESCRIPTION
**Pull request by micheleRP**
fixed https://github.com/redpanda-data/documentation-private/issues/1997

This adds info about how Console automatically tries to deserialize messages and provides troubleshooting info if it finds errors. Or, you can manually select your preferred deserializer. It also moves the deserialization content together under Manage/Redpanda Console.

Preview: 
- https://deploy-preview-175--redpanda-docs-preview.netlify.app/23.3/reference/console/record-deserialization/
- https://deploy-preview-175--redpanda-docs-preview.netlify.app/23.3/manage/console/

### CODE REVIEW

**Comment by micheleRP** [comment]
@weeco, @vladoschreiner: As part of this 23.3 Beta update, I renamed this page and moved it, along with the push filters page, from [Reference/Redpanda Console](https://deploy-preview-175--redpanda-docs-preview.netlify.app/current/reference/console/) to [Manage/Schema Registry](https://deploy-preview-175--redpanda-docs-preview.netlify.app/23.3/manage/schema-reg/). Do you agree? Or do you think these pages belong in [Manage/Redpanda Console](https://deploy-preview-175--redpanda-docs-preview.netlify.app/23.3/manage/console/). 

Also, should the _Protobuf Deserialization_ page be moved from a standalone topic into a subtopic on this _Deserialization in Redpanda Console_ page?

**Comment by micheleRP** [comment]
@weeco, @vladoschreiner: This page seemed to use _message_ and _record_ interchangeable. For consistency, I edited them all to _message_, because of the **Message** tab in the UI. Is that correct?

**Comment by weeco** [comment]
Record is the term that is commonly used in the Kafka terminology. Personally I find "Records" for the tab in the UI less descriptive, but we use it for "Publish Records", "Delete records" etc.

File: modules/reference/pages/console/record-deserialization.adoc [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
-available deserialization methods. The supported deserialization methods are:
+In Redpanda, the messages exchanged between producers and consumers contain raw bytes. Schemas work as an agreed-upon format, like a contract, for producers and consumers to serialize and deserialize those messages. If a producer breaks this contract, consumers can fail. 
 
-* Kafka's internal binary formats (for example the `__consumer_offsets` topic )
+Redpanda Console automatically tries to deserialize incoming messages and show them in human-readable format. It tries different deserializers and lets you choose the deserializer before producing the message in the desired format. If the deserialization fails, it provides troubleshooting information. For example, deserialization requires fetching schemas from a schema registry, and that fetch could fail. If downstream applications crash, you can find which message caused the crash, check the message content, and see the schema that was used in deserializing it. You can also download the raw bytes of the message to feed it directly to your client deserializer or share it with support.
```
Comment (weeco) [inline modules/reference/pages/console/record-deserialization.adoc]: > If the deserialization fails, it provides troubleshooting information.

Maybe worth to clarify what failing here means. As described we try a bunch of deserialization strategies in a specific order until we found a strategy that doesn't yield an error. The very last strategy is called "binary" and in that case we would show the bytes as they are stored in a hex viewer. Sometimes it may be expected to see the payload just in hex bytes (e.g. because the payload is encrypted or because it uses a serializer that we don't support to deserialize). In that case we always show the troubleshoot report though.

Comment (weeco) [reply modules/reference/pages/console/record-deserialization.adoc]: > If downstream applications crash, you can find which message caused the crash, check the message content, and see the schema that was used in deserializing it.

I think this may be a bit misleading. Maybe reword to something that expresses that this may be used as a tool to debug applications that consume messages from that topic. I remember that I provided the deserialization issue an example, but this was a very specific example I had experienced at my past co.

File: modules/reference/pages/console/record-deserialization.adoc [hunk 2]
```diff
@@ -1,27 +1,35 @@
...
-* JSON Schema with schema registry encoding
+* JSON with schema registry encoding
 * XML
 * Avro with schema registry encoding
 * Protobuf
 * Protobuf with schema registry encoding
-* Messagepack (only for topics that are explicitly enabled to test MessagePack)
-* UTF-8 / Strings
+* Messagepack (for topics explicitly enabled to test MessagePack)
+* UTF-8 / strings
```
Comment (weeco) [inline modules/reference/pages/console/record-deserialization.adoc:25]: Smile and uint (8, 16, 32 and 64bit) are missing on that list

File: modules/reference/pages/console/record-deserialization.adoc [hunk 3]
```diff
@@ -1,27 +1,35 @@
...
 
-Decoded messages will be rendered as JSON objects and can also be used as JavaScript object in
+Decoded messages are rendered as JSON objects and can also be used as JavaScript objects in
 xref:./programmable-push-filters.adoc[push filters].
-If none of these deserialization attempts appear to be successful, Redpanda Console will render the byte array
+If no deserialization attempts are successful, Redpanda Console renders the byte array
 in a hex viewer. Encoding formats that are not self-contained requires additional configuration. 
 
+== Manually deserialize messages
```
Comment (weeco) [inline modules/reference/pages/console/record-deserialization.adoc]:
```suggestion
== Specify Deserialization Strategy
```

File: modules/reference/pages/console/record-deserialization.adoc [hunk 4]
```diff
@@ -1,27 +1,35 @@
...
+
+If you don't want to rely on automatic deserialization, or if the automatic deserialization fails, you can manually specify your preferred deserialization format. This can be especially helpful when using plain numbers (for example, int or int32) as the type for key/value, because it might not show up as a number.
```
Comment (weeco) [inline modules/reference/pages/console/record-deserialization.adoc]: Maybe something along those lines: In situations where automatic deserialization may not be reliable or has failed, users have the option to specify their desired deserialization format.

Only slightly different of course and a NIT

**Comment by weeco** [comment]
I don't think this page belongs to the schema registry section. In fact only few deserialization methods involve the schema registry. I think it belongs to Manage > Redpanda Console

> Also, should the Protobuf Deserialization page be moved from a standalone topic into a subtopic on this Deserialization in Redpanda Console page?

Yes!

File: modules/reference/pages/console/record-deserialization.adoc (hunk 1 above)
Comment (micheleRP) [reply modules/reference/pages/console/record-deserialization.adoc]: Thanks @weeco. Please review this updated section: https://deploy-preview-175--redpanda-docs-preview.netlify.app/23.3/reference/console/record-deserialization/

File: modules/reference/pages/console/record-deserialization.adoc (hunk 2 above)
Comment (micheleRP) [reply modules/reference/pages/console/record-deserialization.adoc:25]: Thanks @weeco! Please confirm if this is correct: https://deploy-preview-175--redpanda-docs-preview.netlify.app/23.3/reference/console/record-deserialization/#automatic-deserialization

File: modules/reference/pages/console/record-deserialization.adoc (hunk 3 above)
Comment (micheleRP) [reply modules/reference/pages/console/record-deserialization.adoc]: Since this is a small amount of content I moved it into a Note.

**Review by greg-szrama — APPROVED** [review]
Verbiage looks clean, I'll re-check after you account for Martin's feedback.

File: modules/reference/pages/console/record-deserialization.adoc (hunk 2 above)
Comment (weeco) [reply modules/reference/pages/console/record-deserialization.adoc:25]: Yes looks good to me 👍

File: modules/reference/pages/console/record-deserialization.adoc [hunk 5]
```diff
@@ -1,27 +1,171 @@
...
+In Redpanda, the messages exchanged between producers and consumers contain raw bytes. Schemas work as an agreed-upon format, like a contract, for producers and consumers to serialize and deserialize those messages. If a producer breaks this contract, consumers can fail. 
 
-* Kafka's internal binary formats (for example the `__consumer_offsets` topic )
+Redpanda Console automatically tries to deserialize incoming messages and show them in human-readable format. It tests different deserialization strategies until it finds one with no errors. If no deserialization attempts are successful, Redpanda Console renders the byte array in a hex viewer. Sometimes it may be expected to see the payload in hex bytes (for example, because the payload is encrypted or because it uses a serializer that Redpanda Console cannot deserialize). When this happens, Redpanda Console displays troubleshooting information. This can be a helpful tool to debug applications that consume messages from that topic. You can also download the raw bytes of the message to feed it directly to your client deserializer or share it with support.
```
Comment (weeco) [inline modules/reference/pages/console/record-deserialization.adoc]: Sorry for being picky and possibly confusing before with my other comments, but I would just remove this sentence I believe because I still think it may be confusing to some:

> This can be a helpful tool to debug applications that consume messages from that topic.

**Review by weeco — APPROVED** [review]
Left only two comments, but otherwise looks good to me. Thanks

**Comment by weeco** [comment]
Minor: Would it make sense to remove the "in Redpanda Console" part in the navigation but keep it in the title of the respective document? It feels a bit repetitive:

ss-su2F3jL1

**Comment by micheleRP** [comment]
> Minor: Would it make sense to remove the "in Redpanda Console" part in the navigation but keep it in the title of the respective document? It feels a bit repetitive:
> 
> ss-su2F3jL1

@weeco It does look redundant in the side nav, but I think it's important to keep Console in the page title for better search results (e.g., _Use the Schema Registry API_ vs _Use Schema Registry in Redpanda Console_).

**Review by vladoschreiner — APPROVED** [review]
LGTM

File: modules/reference/pages/console/record-deserialization.adoc [hunk 6]
```diff
@@ -1,27 +1,171 @@
...
+requires Redpanda Console to be aware of the used `.proto` files as well as a mapping of what prototype should be used for each topic.
+This information can either be sourced from the Schema Registry or it can be provided with additional configuration so the files
+can be pulled from the local file system or a GitHub repository.
+
+NOTE: To support imports, all prototypes are first registered in a proto registry.
+You must ensure that all imported prototypes are part of the repository. Standard types (such as Google's timestamp type) are included by default and don't need to be added.
+
+=== Schema Registry
+
+Messages that have been serialized using Confluent's KafkaProtobufSerializer can only be deserialized if the Schema Registry is configured.
```
Comment (greg-szrama) [inline modules/reference/pages/console/record-deserialization.adoc]:
```suggestion
Messages that have been serialized using Confluent's KafkaProtobufSerializer can only be deserialized if the Schema Registry is configured to recognize them.
```

I feel like this sentence needs closure so the reader isn't left hanging.

File: modules/reference/pages/console/record-deserialization.adoc [hunk 7]
```diff
@@ -1,27 +1,171 @@
...
+    fileSystem:
+      enabled: true
+      refreshInterval: 5m # 5min is the default refresh interval
+      paths:
+        - /etc/protos
+----
+
+=== GitHub repository
+
+To provide the proto files through a GitHub repository, first place the files in a directory. The directory doesn't matter, because
```
Comment (greg-szrama) [inline modules/reference/pages/console/record-deserialization.adoc]: You might want to  rephrase this -- "Redpanda will search the specified GitHub repository to a directory depth of up to five levels, so no specific directory name is required."

File: modules/reference/pages/console/record-deserialization.adoc [hunk 8]
```diff
@@ -1,27 +1,171 @@
...
+        url: https://github.com/redpanda-data/owlshop-protos.git
+      basicAuth:
+        enabled: true
+        username: token # API token from basic auth
+        password: redacted
+----
+
+=== Topic mapping
+
+If you don't use the Schema Registry for Protobuf deserialization, you must provide a mapping configuration, so Redpanda Console is aware of what
```
Comment (greg-szrama) [inline modules/reference/pages/console/record-deserialization.adoc]:
```suggestion
If you don't use the Schema Registry for Protobuf deserialization, you must provide a mapping configuration so Redpanda Console is aware of what
```

**Review by greg-szrama — APPROVED** [review]
Just a few comments and suggestions to improve clarity. Content looks great!
